common/errorinjector/noop_impl.go

```diff
@@ -0,0 +1,18 @@
+//go:build !errorinjector
```

Comment:
I'd like to consider flipping this, actually. A while ago I experimented with build tags and assertions; and I feel that confident that we can add a step for the actual binary build that enforces verifies no trace of `ErrorInjector` can be found as it was optimized away. That way we don't need to tell every single developer to run their tests with this flag.

Reply:
I like that, but I'm still a little concerned since we have so many different binary builds (docker images, goreleaser, internal ones), and then users that build their own binaries...

Reply:
FWIW, docker-builds seems to already [invoke the server Makefile](https://github.com/temporalio/docker-builds/blob/4af24b21018ad1e0a932db3135356ea1e0834e12/Makefile#L50).